modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationEndpoints/EditAPIApplicationEndpoint.testcase

```diff
@@ -35,6 +35,65 @@ definition {
 		}
 	}
 
+	@priority = 3
+	test CanChangeSchemaOfEndpoint {
+		task ("And Given two API Schema in API Application") {
+			SchemaAPI.createAPISchema(
+				apiApplicationERC = "My-app",
+				mainObjectDefinitionERC = "L_API_APPLICATION",
+				name = "schema1");
+
+			SchemaAPI.createAPISchema(
+				apiApplicationERC = "My-app",
+				mainObjectDefinitionERC = "L_API_APPLICATION",
+				name = "schema2");
+		}
+
+		task ("And Given setting API Schema schema1 in Response Body Schema") {
+			APIBuilderUI.addAndPublishEndpointConfiguration(
+				entity = "Endpoints",
+				name = "testendpoint",
+				schemaName = "schema1",
+				title = "My-app");
+
+			AssertElementPresent(
+				key_optionName = "schema1",
+				locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+		}
+
+		task ("When modifying endpoint again with API Schema schema2") {
+			APIBuilderUI.addAndPublishEndpointConfiguration(
+				entity = "Endpoints",
+				name = "testendpoint",
+				schemaName = "schema2",
+				title = "My-app");
+		}
+
+		task ("Then changes are saved correctly") {
+			Alert.viewSuccessMessage();
+
+			AssertElementPresent(
+				key_optionName = "schema2",
+				locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+		}
+	}
+
+	@priority = 3
+	test CanCreateEndpointInPublishedAPIApplication {
+		task ("When editing API Application") {
+			APIBuilderUI.switchToRelatedEntryInEditApplication(
```

Comment:
There is no entity with title `test2`, could you check and update? It is better to run the case you added locally and if it passed before send a pr. Here are two aspects to update for this test
- title test2
- We can switch to Endpoint tab and create an endpoint in `task ("When editing API Application"){}`, and then assert the new endpoint is actually created in `task ("Then API Endpoint can be created"){}`